src/sentry/lang/native/minidump.py

```diff
@@ -54,17 +55,50 @@ def is_minidump_event(data):
     return get_path(exceptions, 0, "mechanism", "type") == "minidump"
 
 
-def merge_attached_event(mpack_event, data):
+def decode_payload(encoded_data, max_size=None, multiple=False):
```

Comment:
Generally nice, but `multiple=True` for JSON is very strange. Can we instead ignore `multiple` for JSON and just make sure that the SDK serializes a list instead of an object? I'm slightly uncomfortable with supporting mashed JSON files that are not valid JSON...

I'm always fuzzy on the details, but doesn't `encoded_data.readline()` allocate the entire buffer redundantly just to check the first byte? We could just read the first byte and inspect that, then stream into the `Unpacker`.

Reply:
Or better than ignoring `multiple`, check that the output is a list.

Reply:
I don't like the fact that JSON would have a different format than msgpack here. That makes the formats not interchangable. Multi JSON payloads are a common system all over the place.

As per readline in python that optimization makes very little sense. The overhead of a wrapper stream is likely to be higher than that little memcpy in c.